pkg/config/config.go

```diff
@@ -377,8 +434,103 @@ func processConfigFile(
+		}
+
+		var resolvedPaths []string
+		var err error
+
+		if strings.HasPrefix(importPath, "http://") || strings.HasPrefix(importPath, "https://") {
+			// Handle remote URLs
+			// Validate the URL before downloading
+			parsedURL, err := url.Parse(importPath)
+			if err != nil || (parsedURL.Scheme != "http" && parsedURL.Scheme != "https") {
+				u.LogWarning(atmosConfig, fmt.Sprintf("unsupported URL '%s': %v", importPath, err))
+				continue
+			}
+
+			tempDir, tempFile, err := downloadRemoteConfig(importPath)
+			if err != nil {
+				u.LogWarning(atmosConfig, fmt.Sprintf("failed to download remote config '%s': %v", importPath, err))
+				continue
+			}
+			resolvedPaths = []string{tempFile}
+			defer os.RemoveAll(tempDir)
+
+		} else {
+
+			ext := filepath.Ext(importPath)
+
+			basePath, err := filepath.Abs(atmosConfig.BasePath)
+			if err != nil {
+				return err
+			}
+			if ext != "" {
+				imp := filepath.Join(basePath, importPath)
+				resolvedPaths, err = u.GetGlobMatches(imp)
+				if err != nil {
+					u.LogWarning(atmosConfig, fmt.Sprintf("failed to resolve import path '%s': %v", imp, err))
+					continue
+				}
+			} else {
+				impYaml := filepath.Join(basePath, importPath+".yaml")
+				impYml := filepath.Join(basePath, importPath+".yml")
+				// ensure the joined path doesn't escape the intended directory
+				if !strings.HasPrefix(impYaml, basePath) || !strings.HasPrefix(impYml, basePath) {
+					return fmt.Errorf("invalid import path: attempted directory traversal")
+				}
+				resolvedPathYaml, errYaml := u.GetGlobMatches(impYaml)
+				resolvedPathsYml, errYml := u.GetGlobMatches(impYml)
+				if errYaml != nil && errYml != nil {
+					u.LogWarning(atmosConfig, fmt.Sprintf("failed to resolve import path '%s': %v", importPath, err))
+					continue
+				}
+				resolvedPaths = append(resolvedPaths, resolvedPathYaml...)
+				resolvedPaths = append(resolvedPaths, resolvedPathsYml...)
+
+			}
+		}
+		// print the resolved paths
+		u.LogTrace(atmosConfig, fmt.Sprintf("Resolved import paths: %v", resolvedPaths))
+		for _, path := range resolvedPaths {
+			// Process each configuration file
+			_, _, err = processConfigFile(atmosConfig, path, v)
+			if err != nil {
+				// Log the error but continue processing other files
+				u.LogWarning(atmosConfig, fmt.Sprintf("failed to merge configuration from '%s': %v", path, err))
+				continue
+			}
+		}
 	}
+	return nil
+}
 
-	return true, nil
+func downloadRemoteConfig(url string) (string, string, error) {
+	tempDir, err := os.MkdirTemp("", "atmos-import-*")
+	if err != nil {
+		return "", "", err
+	}
+	tempFile := filepath.Join(tempDir, "atmos.yaml")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	client := &getter.Client{
+		Ctx:  ctx,
+		Src:  url,
+		Dst:  tempFile,
+		Mode: getter.ClientModeFile,
+	}
+	err = client.Get()
+	if err != nil {
+		os.RemoveAll(tempDir)
+		return "", "", fmt.Errorf("failed to download remote config: %w", err)
```

Comment:
I'm concerned if we're repeating calls to GoGetter, since we have another implementation that properly handles `GITHUB_TOKEN`. Do not want that logic distributed.